src/tasks/submit/flashbots.rs

```diff
@@ -117,38 +117,54 @@ impl FlashbotsTask {
                 debug!("upstream task gone - exiting flashbots task");
                 break;
             };
-            let span = sim_result.span();
-            span_debug!(span, "received sim result");
+
+            let span = sim_result.sim_env.clone_span();
+
+            // Don't submit empty blocks
+            if sim_result.block.is_empty() {
+                counter!("signet.builder.flashbots.empty_block").increment(1);
+                span_debug!(span, "received empty block - skipping");
+                continue;
+            }
+            span_debug!(span, "flashbots task received block");
 
             // Prepare a MEV bundle with the configured call type from the sim result
-            let Ok(bundle) =
-                self.prepare(&sim_result).instrument(span.clone()).await.inspect_err(|error| {
-                    counter!("signet.builder.flashbots.bundle_prep_failures").increment(1);
-                    span_debug!(span, %error, "bundle preparation failed");
-                })
-            else {
+            let res = self.prepare(&sim_result).instrument(span.clone()).await;
```

Comment:
this block is less legible than it should be. either revert to the way it was, or use a match statement